### PR TITLE
added file upload to amazon s3

### PR DESCRIPTION
**Pull request by lamuria**
Added file upload to amazon s3 and get file url functions.
With this new functions we are able to upload any kind of files to s3.

Usage:

```
  Inputs: - credentials {:bucket
                         :access-key
                         :secret}
          - file-key We ensure an unique file key by concating an uuid, if nil we generate an unique file-key.
          - file-value {:value ;file-content
                        :content-type }

  Returns: file url to s3

  (file-management/put-file credentials-hash file-key file-info)
```

ref: https://github.com/clanhr/mothership/issues/1266

### CODE REVIEW

**Comment by donbonifacio** [comment]
Is the ref correct?

:ok: 

Could have a better commit, and also some docstrings on those fns. Also, do we have tests for this?

**Comment by lamuria** [comment]
@donbonifacio What kind of tests are you talking about?

**Comment by donbonifacio** [comment]
@lamuria the actual send something to s3.

File: src/file_management/core.clj [hunk 1]
```diff
@@ -48,3 +56,26 @@
                    {:content-type "image/png"}
                    (s3/grant :all-users :read))
     (get-avatar-url credentials user-id))
+
+(defn gets-file-key
+  [file-key]
+  (str file-key "-" (java.util.UUID/randomUUID)))
```
Comment (donbonifacio) [inline src/file_management/core.clj]: On your commit message you say that if the file-key exists it will be smashed. Don't this line prevent it?

Comment (lamuria) [inline src/file_management/core.clj]: yep! good point! This was done after I wrote the commit message..

**Pull request merged**